swagger/sales/coupons.json

```diff
@@ -0,0 +1,96 @@
+  "info": {
+    "title": "API V3",
+    "version": "v3.0"
+  },
+  "paths": {
+    "/v3/sales/coupons": {
+      "get": {
+        "summary": "Get coupons api",
+        "description": "Get list of coupons",
+        "x-amazon-apigateway-integration": {
+          "responses": {
+            "default": {
+              "statusCode": "200"
+            }
+          },
+          "uri": "https://http://localhost:7100/v3/sales/coupons",
+          "httpMethod": "GET",
+          "type": "http_proxy"
+        },
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "translation missing: en.sales-api.coupons_message.tag"
+        ],
+        "security": [
+          {
+            "default": [
+              "business/payments"
+            ]
+          }
+        ],
+        "parameters": [
+          {
+            "name": "Authorization",
+            "in": "header",
+            "type": "string",
+            "required": true,
+            "description": "The format of the API authorization header should be: 'Authorization': \"Bearer API_TOKEN\"",
+            "x-example": "Bearer abc...123"
+          }
+        ],
+        "x-amazon-apigateway-auth": {
+          "type": "NONE"
+        },
+        "x-auth-type": "Application & Application User",
+        "responses": {
+          "200": {
+            "description": "Success",
+            "examples": {
+              "application/json": {
+                "success": true,
+                "data": {
+                  "success": true,
+                  "data": {
+                    "success": "true / false",
+                    "data": [
+                      {
+                        "uid": "wo9jvxhpb9fqmrqp",
+                        "business_uid": "k72pvtwaui4pa05v",
+                        "coupon_name": "General 20% OFF coupon",
+                        "discount_type": "percent",
+                        "discount_value": 20.0,
+                        "coupon_code": "SNT6CDBWKY",
+                        "status": "active",
+                        "valid_for_services": [
+                          "service_uid1"
+                        ],
+                        "valid_for_staff": [
+                          "staff_uid1"
+                        ],
+                        "start_date": "2024-04-21T11:35:52.000+03:00",
+                        "expiration_date": "2025-04-21T11:35:52.000+03:00",
+                        "max_redemptions": 25,
+                        "max_redemptions_per_client": 1,
+                        "total_redemptions": 0,
+                        "created_at": "2024-04-21T11:35:52.000+03:00",
+                        "updated_at": "2024-04-21T11:35:52.000+03:00"
+                      }
+                    ]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
```

Comment:
Note the return format is wrong. Have a look in access_control or in bizai.
Are you only enabling get list? what about the rest of the CRUD?

Reply:
at the moment it's only the get list